src/oscar/apps/customer/forms.py

```diff
@@ -226,14 +227,19 @@ def _orders_description(self, date_from, date_to, order_number):
     def get_filters(self):
         date_from = self.cleaned_data['date_from']
         date_to = self.cleaned_data['date_to']
+
         order_number = self.cleaned_data['order_number']
         kwargs = {}
-        if date_from and date_to:
-            kwargs['date_placed__range'] = [date_from, date_to]
-        elif date_from and not date_to:
-            kwargs['date_placed__gt'] = date_from
-        elif not date_from and date_to:
+
+        if date_from:
+            kwargs['date_placed__gte'] = date_from
+
+        if date_to:
+            # A day is added because date_to is inclusive so that if you search for orders placed from 1st to 3rd you
+            # will get orders placed on 1st 2nd and 3rd.
+            date_to += timedelta(days=1)
```

Comment:
I think it would be preferable to use the same approach as [we do in the dashboard](https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/dashboard/orders/views.py#L190-L191).